### PR TITLE
scotch: switch to cmake

### PR DESCRIPTION
**Pull request by 3rav**

### CODE REVIEW

File: mingw-w64-scotch/PKGBUILD [hunk 1]
```diff
@@ -33,114 +35,84 @@ prepare() {
...
 }
 
-package_scotch-int64() {
+package_scotch64() {
   pkgdesc="Graph partitioning and sparse matrix ordering package with 64-bit indexing (mingw-w64)"
-  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-  _package_scotch "64"
+  provides=()
+  conflicts=()
```
Comment (mmuetzel) [inline mingw-w64-scotch/PKGBUILD:106]: Should this replace the package with the previous name?
``` suggestion
  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}-int64")
  replaces=("${MINGW_PACKAGE_PREFIX}-${_realname}-int64")
```

File: mingw-w64-scotch/PKGBUILD [hunk 2]
```diff
@@ -33,114 +35,84 @@ prepare() {
...
+  
+  # Static Install
+  cd "${srcdir}"/build-${MSYSTEM}-static-64
+  DESTDIR=${pkgdir} cmake --install . 
+  
+  # Shared Install
+  cd "${srcdir}"/build-${MSYSTEM}-shared-64
+  DESTDIR=${pkgdir} cmake --install . 
+  install -Dm644 "${srcdir}"/${_realname}-v${pkgver}/doc/CeCILL-C_V1-en.txt "${pkgdir}"${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE
```
Comment (mmuetzel) [inline mingw-w64-scotch/PKGBUILD:115]: If the packages with 64-bit and 32-bit indices should be installable in parallel, you'd probably need to install the license somewhere else. E.g.:
``` suggestion
  install -Dm644 "${srcdir}"/${_realname}-v${pkgver}/doc/CeCILL-C_V1-en.txt "${pkgdir}"${MINGW_PREFIX}/share/licenses/${_realname}64/LICENSE
```

**Comment by mmuetzel** [comment]
If you'll actually like to have packages that don't conflict, you'll probably have to change the `CMakeLists.txt` files.
E.g., for the libraries, you could add something like `set_target_properties(${_target} PROPERTIES OUTPUT_NAME ${_target}${NAME_SUFFIX})` at around here: 
https://gitlab.inria.fr/scotch/scotch/-/blob/master/src/libscotch/CMakeLists.txt?ref_type=heads#L825

And then set, e.g., `-DNAME_SUFFIX=64` when building the 64-bit index version.
But it would probably be best to coordinate that with the upstream developers.

**Comment by 3rav** [comment]
1. `.dll` library files are not installed to the BIN directory, even though they are in the folders: build-MINGW64-shared-32/bin
2. error, when linking to it own library:
```
[244/520] Linking C shared library bin\libscotch.dll
FAILED: bin/libscotch.dll lib/libscotch.dll.a 
C:\Windows\system32\cmd.exe /C "cd . && D:\M\msys64\mingw64\bin\gcc.exe -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wp,-D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG   -shared -o bin\libscotch.dll -Wl,--out-implib,lib\libscotch.dll.a -Wl,--major-image-version,7,--minor-image-version,0 @CMakeFiles\scotch.rsp && cd ."
D:/M/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/libscotch/CMakeFiles/scotch.dir/arch.c.obj:arch.c:(.text+0x220): undefined reference to `SCOTCH_errorPrint'
```
3. error in clang:
```
[6/520] Building C object src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj
  FAILED: src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj 
  D:\M\msys64\clang64\bin\clang.exe -DCOMMON_OS_WINDOWS -DCOMMON_PTHREAD -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_THREAD_WIN32 -DINTSIZE32 -DSCOTCH_PATCHLEVEL_NUM=5 -DSCOTCH_PTHREAD -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_RENAME -DSCOTCH_VERSION_NUM=7 -Drestrict=__restrict  -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wp,-D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -MD -MT src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj -MF src\libscotch\CMakeFiles\dummysizes.dir\dummysizes.c.obj.d -o src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj -c C:/_/B/src/scotch-v7.0.5/src/libscotch/dummysizes.c
  In file included from C:/_/B/src/scotch-v7.0.5/src/libscotch/dummysizes.c:94:
  In file included from C:/_/B/src/scotch-v7.0.5/src/libscotch/common.h:153:
  C:/_/B/src/scotch-v7.0.5/src/libscotch/pthread_win32.h:87:13: error: use of undeclared identifier 'EINVAL'
     87 |     return (EINVAL);
        |             ^
  C:/_/B/src/scotch-v7.0.5/src/libscotch/pthread_win32.h:91:13: error: use of undeclared identifier 'EAGAIN'
     91 |     return (EAGAIN);
```

**Comment by 3rav** [comment]
@Biswa96 please take a look at this

**Comment by 3rav** [comment]
Fix:
1. .dll library files are not installed to the BIN directory, even though they are in the folders: build-MINGW64-shared-32/bin
2. error, when linking to it own library -lscotcherr. by adapting fix from: https://gitlab.inria.fr/scotch/scotch/-/issues/16#note_771661

There is still an error in the clang environment:
```
[6/520] Building C object src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj
  FAILED: src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj 
  D:\M\msys64\clang64\bin\clang.exe -DCOMMON_OS_WINDOWS -DCOMMON_PTHREAD -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_THREAD_WIN32 -DINTSIZE32 -DSCOTCH_PATCHLEVEL_NUM=5 -DSCOTCH_PTHREAD -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_RENAME -DSCOTCH_VERSION_NUM=7 -Drestrict=__restrict  -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wp,-D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -MD -MT src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj -MF src\libscotch\CMakeFiles\dummysizes.dir\dummysizes.c.obj.d -o src/libscotch/CMakeFiles/dummysizes.dir/dummysizes.c.obj -c C:/_/B/src/scotch-v7.0.5/src/libscotch/dummysizes.c
  In file included from C:/_/B/src/scotch-v7.0.5/src/libscotch/dummysizes.c:94:
  In file included from C:/_/B/src/scotch-v7.0.5/src/libscotch/common.h:153:
  C:/_/B/src/scotch-v7.0.5/src/libscotch/pthread_win32.h:87:13: error: use of undeclared identifier 'EINVAL'
     87 |     return (EINVAL);
        |             ^
  C:/_/B/src/scotch-v7.0.5/src/libscotch/pthread_win32.h:91:13: error: use of undeclared identifier 'EAGAIN'
     91 |     return (EAGAIN);
```

**Pull request closed without merging**